src/main/java/org/opensearch/neuralsearch/processor/InferenceProcessor.java

```diff
@@ -84,6 +88,7 @@ public InferenceProcessor(
         this.mlCommonsClientAccessor = clientAccessor;
         this.environment = environment;
         this.clusterService = clusterService;
+        mappingForNestedFields = fieldMap.entrySet().stream().map(this::processNestedKey).collect(Collectors.toList());
```

Comment:
I didn't mean to just move this in constructor. As we are calling processNestedKey inside execute method anyway, there is no much difference between this code and previous code. Actually, previous code is better...

Reply:
got it, yeah there is no easy way to remove that call from recursion. reverting back to the previous state